File: Maximum-Gap/MG.rs
pub fn maximum_gap(nums: Vec<i32>) -> i32 {
    let length = nums.len();
    if length < 2 as usize {
        return 0;
    }

    let mut new_nums = nums.clone();
    new_nums.sort();
    let mut largest = 0;
    for i in 1..length {
        if new_nums[i] - new_nums[i - 1] > largest {
            largest = new_nums[i] - new_nums[i - 1];
        }
    }

    largest
}

fn main() {
    println!("{:?}", maximum_gap(vec![3, 6, 9, 1])); //=> 3
    println!("{:?}", maximum_gap(vec![10])); //=> 0
}

File: Maximum-Gap/MG.jl
function maximumGap(nums)
    length(nums) < 2 && return 0

    sort!(nums)
    largest = 0;
    last = nums[1]
    for c in nums
        (c - last) > largest && begin largest = (c - last) end
        last = c
    end

    largest
end

using Test
@test maximumGap([3,6,9,1]) == 3
@test maximumGap([10]) == 0
